packages/env-tests/src/tests/exchange.ts

```diff
@@ -11,64 +18,70 @@ export function runExchangeTest(context: EnvTestContext) {
       await fundAccount(context, TestAccounts.Exchange, ONE.times(10))
     })
 
-    test('exchange cUSD for CELO', async () => {
-      const from = await getKey(context.mnemonic, TestAccounts.Exchange)
-      context.kit.connection.addAccount(from.privateKey)
-      context.kit.defaultAccount = from.address
-      const stableToken = await context.kit.contracts.getStableToken()
-      context.kit.connection.defaultFeeCurrency = stableToken.address
-      const goldToken = await context.kit.contracts.getGoldToken()
-      const exchange = await context.kit.contracts.getExchange()
+    for (const stableToken of context.stableTokensToTest) {
+      test(`exchange ${stableToken} for CELO`, async () => {
+        let stableTokenInstance = await initStableTokenFromRegistry(stableToken, context.kit)
 
-      const previousGoldBalance = await goldToken.balanceOf(from.address)
-      const goldAmount = await exchange.quoteUsdSell(ONE)
-      logger.debug({ rate: goldAmount.toString() }, 'quote selling cUSD')
+        const from = await getKey(context.mnemonic, TestAccounts.Exchange)
+        context.kit.connection.addAccount(from.privateKey)
+        context.kit.defaultAccount = from.address
+        context.kit.connection.defaultFeeCurrency = stableTokenInstance.address
+        const goldToken = await context.kit.contracts.getGoldToken()
 
-      const approveTx = await stableToken.approve(exchange.address, ONE.toString()).send()
-      await approveTx.waitReceipt()
-      const sellTx = await exchange
-        .sellDollar(
-          ONE,
-          // Allow 5% deviation from the quoted price
-          goldAmount
-            .times(0.95)
-            .integerValue(BigNumber.ROUND_DOWN)
-            .toString()
-        )
-        .send()
-      await sellTx.getHash()
-      const receipt = await sellTx.waitReceipt()
+        let exchange = await initExchangeFromRegistry(stableToken, context.kit)
+        const previousGoldBalance = await goldToken.balanceOf(from.address)
+        const goldAmount = await exchange.getBuyTokenAmount(ONE, false)
+        logger.debug({ rate: goldAmount.toString() }, `quote selling ${stableToken}`)
```

Comment:
Also amend the logger with the stabletoken info

Reply:
Done!